src/@core/src/genre/domain/entities/genre.ts

```diff
@@ -55,26 +56,43 @@ export class Genre extends AggregateRoot<
     this.name = name;
   }
 
-  addCategoryId(categoryId: CategoryId): void {
-    this.categories_id.set(categoryId.value, categoryId);
+  addCategoryId(categoryId: CategoryId) {
+    const categoriesId = cloneDeep(this.props.categories_id);
+    categoriesId.set(categoryId.value, categoryId);
+
+    Genre.validate({
+      ...this.props,
+      categories_id: categoriesId,
+    });
+
+    this.categories_id = categoriesId;
```

Comment:
Podemos aplicar a validação de dados antes de realizar a atribuição sem usar o cloneDeep, exemplo:
```typescript
Genre.validate({
  ...this.props,
  categories_id: new Map(this.categories_id).set(categoryId.value, categoryId)
})
```
Um map pode ser criado através de outro map e neste caso criamos um novo map com os dados do antigo e adicionando o novo elemento para ser validado.

Reply:
Precisamos do valor validado para atribuir à propriedade.
```suggestion
   addCategoryId(categoryId: CategoryId) {
    const categoriesId = new Map(this.categories_id).set(categoryId.value, categoryId);

    Genre.validate({
      ...this.props,
      categories_id: categoriesId,
    });

    this.categories_id = categoriesId;
  }
```